measureme/src/counters.rs

```diff
@@ -572,42 +549,49 @@ mod hw {
     /// only requires one "serializing instruction", rather than two.
     #[inline(always)]
     fn rdpmc_pair(a_reg_idx: u32, b_reg_idx: u32) -> (u64, u64) {
-        let (a_lo, a_hi): (u32, u32);
-        let (b_lo, b_hi): (u32, u32);
+        serialize_instruction_execution();
+        (unserialized_rdpmc(a_reg_idx), unserialized_rdpmc(b_reg_idx))
+    }
+
+    /// Dummy `cpuid(0)` to serialize instruction execution.
+    #[inline(always)]
+    fn serialize_instruction_execution() {
```

Comment:
Should be able to take this `fn` in isolation and try it on godbolt, for the comparison with `std::arch::x86_64::__cpuid(0);`.

Reply:
minor difference with using arch version one extra xor and xchg instead of the mov that I used.

https://godbolt.org/z/8jv41bYsq